pkg/indexer/indexer.go

```diff
@@ -0,0 +1,418 @@
+package indexer
+
+import (
+	"path/filepath"
+	"time"
+
+	"github.com/pkg/errors"
+	"gorm.io/driver/sqlite"
+	"gorm.io/gorm"
+	"gorm.io/gorm/clause"
+
+	"github.com/gohornet/hornet/pkg/model/milestone"
+	"github.com/gohornet/hornet/pkg/model/utxo"
+	"github.com/gohornet/hornet/pkg/whiteflag"
+	"github.com/iotaledger/hive.go/kvstore/utils"
+	iotago "github.com/iotaledger/iota.go/v3"
+)
+
+var (
+	ErrNotFound = errors.New("output not found for given filter")
+
+	tables = []interface{}{
+		&status{},
+		&extendedOutput{},
+		&nft{},
+		&foundry{},
+		&alias{},
+	}
+)
+
+type Indexer struct {
+	db *gorm.DB
+}
+
+func NewIndexer(dbPath string) (*Indexer, error) {
+
+	if err := utils.CreateDirectory(dbPath, 0700); err != nil {
```

Comment:
Should we also use our `CheckDatabaseEngine` function here, so we put the `dbInfo` file inside the folder, in case we support other engines for the indexer in the future as well?

Reply:
Not sure this makes sense in this case. We are constructing queries that are not guaranteed to work with any other database engine right now.

Reply:
I added #1283 for this